### DISCUSSION TITLE
Wow we would ideally communicate about infrastructure and support

### DISCUSSION THREAD

**Discussion by mattjbr123** [ukceh-rse/community#12] [Science Computing Group issues]
Typo kept for fun xD

We have had some feedback lately that it is difficult to get a full picture of the computing infrastructure estate at UKCEH. There is some information around but it seems to be scattered on old Sharepoint pages or in private wikis within wiki.ceh.ac.uk that are obscure and unclear how to get access to or what is even there.
It feels like there must be a better way that we can collate and display this information, as well as who is responsible for support for each element of the infrastructure too (a common complaint is "I emailed some person about X and they told me they weren't responsible and passed me on to a different person and so on")

**Comment by fsamreen** [comment]
Brain dump - a structured list of items/information would be helpful (published through various platforms - SharePoint, Hub, etc):

- **Infrastructure list** 
- [ ] A clear overview of available infrastructure and its capacity, described in a way that’s **_user-friendly and easy to interpret_** (avoid jargons and acronyms).
- **Resource request process** 
- [ ] A defined process for requesting infrastructure resources. 
- [ ] A standardised form or template specifying the required information for each request (e.g., resource type, intended use, duration, etc.).
- **Software/Tools installation** 
- [ ] Guidelines for requesting additional software or tools, especially relevant when users request virtual machines.
- **Provisioning time** 
- [ ] Estimated time frames for each stage of the provisioning process, including start and expected completion dates.
- **Access and usage documentation** 
- [ ] A detailed guide on accessing and using the resources.
- [ ] Instructions on best practices for documenting usage to ensure clarity and support.
- **Resource termination** 
- [ ] A clear process for reporting when resources are no longer needed, allowing for efficient resource reallocation.
- **Support service** 
- [ ] Contact details to get support or a web-based thing to request support for a service/infrastructure. Estimated time for the support team to pick up the support ticket.
- [ ] FAQs
- [ ] Community group(s)
- [ ] Service availability and updates - ways to communicate this to the service/infrastructure users

**Reply by mattjbr123** [reply]
Thanks Faiza, this is super useful, we can use this as an agenda for the next meeting perhaps

**Comment by samharrison7** [comment]
I definitely support this, and Faiza's list is great. I also think some guidance on *how* we're meant to use VMs would be useful - e.g. should we be treating them as ephemeral and discarding/creating for each new project, or should we be keeping hold of them. If I need more resources, should I request a new VM, or should I request my existing VM is scaled up to match. How long will each of these take to action? Etc...

**Reply by mattjbr123** [reply]
Agreed!

**Comment by Jez-Carter** [comment]
It's cool that this is being discussed! I was at a statistics networking event at UKCEH the other week and this topic was raised. People found it hard to understand what compute resources are available and how to access them. It seems to me that some extra resources on this would be very valuable. I'd potentially extend it to also include basic Git and GitHub information and suggested coding environments (e.g. VS Code rather than doing everything in the terminal), even starting to incorporate stuff like how to make most use of the new AI coding tools. This is all stuff that has plenty of documentation out there but can feel overwhelming and difficult to understand what to focus on. It has a major impact on efficiency and enjoyment of coding though!

**Reply by Jez-Carter** [reply]
I think even just a list of amazing resources that already cover these topic areas would be cool. I love this page by Patrick J Mineault for example: [The good research code handbook](https://goodresearch.dev/index.html). Anyway, apologies if I've extended beyond the scope of this discussion but incidentally if anyone has particular blogs etc that they really love on these topics please let me know and I'd be interested to read through them myself :relaxed:

**Reply by mattjbr123** [reply]
Not at all, thanks for the contribution, I think this is definitely something we can look to develop as part of the remit of the RSE team and in collaboration with IT. It is also useful information to know this is coming up in other areas of the organisation that too - the more it's raised the more we can push for change :D

**Comment by metazool** [comment]
A colleague who's also a member of both IT Engagement Groups offered their blessing to repost this from an internal thread about an infrastructure migration project that's grappling with changing licence conditions:

_As part of this, would also be really good to get it explained as to: a) who manages / will manage the server and whether they get time to do so, b) how many users / apps / apps-per-user / users-per-app are allowed, c) how long requests may/should take to get actioned etc, d) where funding comes from and will come from in the longer term._

To me, some of this is asset registry
1. Single source of truth about maintenance point of contact, available for anyone who wants to look for it
2. Information about systems the service is running on, dependencies, and a funding-related lifespan

And some of it is social support
1. Overview documentation for onboarding that encourages edits from people who've recently onboarded
2. Accessible and clear feedback about how requests for information or action are being handed around

**Comment by metazool** [comment]
Another item surfacing from the IT engagement groups, a research group in Wallingford that have developed their own informal 3-part onboarding course, the first part of which is discovery of available systems (on-prem VMs, HPC, JASMIN, etc), then intro Linux and intro python including environment management

Were we to take this walkthrough documentation, then test, expand and publish it, where would that be?

**Reply by metazool** [reply]
What else would it cover? Access to databases, and to API based services, and where people have already developed client libraries that could be maintained for wider use

**Reply by samharrison7** [reply]
That sounds really useful. Do they make their course materials openly available, i.e. could we have a look?

The Hub seems to be the recommended place for this type of content, but I suspect this wouldn't be flexible enough for such material. I think a nicer solution would be a [Jupyter Book](https://jupyterbook.org/en/stable/intro.html) (or similar) that contains all of the relevant information for using UKCEH science computing infrastructure. Ideally, this would be built from a public repo that everyone can submit PRs to. Initially I guess this would be developed by IT Support, but if it is a public repo then users could supplement the documentation with their own tips and tricks to make it as useful as possible. This Jupyter Book could be linked to from a Hub page. That would be my preference anyway!

**Reply by mattjbr123** [reply]
This does sound like a great place to start - very interested to see what they've included and how!

**Reply by Jez-Carter** [reply]
Sounds good to me. Incidentally just to mention I've been developing a Jupyter book for the 'data science toolbox' deliverable of the NCUK WP2 project. We've been debating about the remit of that Jupyter book, it's mostly focused around presenting statistical methods developed at UKCEH that are generalisable across multiple domains, although we were also wondering about whether or not to incorporate some general guidance around coding practices and computational resources etc. :smiley:
